lightsaber/processor/src/main/java/com/joom/lightsaber/processor/validation/UsageValidator.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 SIA Joom
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.joom.lightsaber.processor.validation
+
+import com.joom.grip.ClassesResult
+import com.joom.grip.Grip
+import com.joom.grip.Query
+import com.joom.grip.annotatedWith
+import com.joom.grip.classes
+import com.joom.grip.mirrors.ClassMirror
+import com.joom.grip.mirrors.Type
+import com.joom.grip.superType
+import com.joom.grip.withField
+import com.joom.grip.withMethod
+import com.joom.lightsaber.LightsaberTypes
+import com.joom.lightsaber.processor.ErrorReporter
+import com.joom.lightsaber.processor.analysis.SourceResolver
+import com.joom.lightsaber.processor.commons.Types
+import com.joom.lightsaber.processor.model.InjectionContext
+import java.nio.file.Path
+
+class UsageValidator(
+  private val grip: Grip,
+  private val errorReporter: ErrorReporter,
+  private val sourceResolver: SourceResolver,
+) {
+
+  fun validateUsage(injectionContext: InjectionContext, paths: Collection<Path>) {
+    val modulesQuery = grip select classes from paths where annotatedWith(Types.MODULE_TYPE)
+    val componentsQuery = grip select classes from paths where annotatedWith(Types.COMPONENT_TYPE)
+    val contractConfigurationsQuery = grip select classes from paths where superType { _, type -> type == Types.CONTRACT_CONFIGURATION_TYPE }
+    val methodsQuery = grip select classes from paths where withField { _, fieldMirror -> fieldMirror.annotations.contains(Types.INJECT_TYPE) }
```

Comment:
Maybe we need to swap the variable names?

Reply:
Good catch! Fixed